cores/esp8266/Esp.h

```diff
@@ -150,8 +150,10 @@ class EspClass {
         bool checkFlashCRC();
 
         bool flashEraseSector(uint32_t sector);
-        bool flashWrite(uint32_t offset, uint32_t *data, size_t size);
+        bool flashWrite(uint32_t offset, const uint32_t *data, size_t size);
+        bool flashWrite(uint32_t offset, const uint8_t *data, size_t size);
         bool flashRead(uint32_t offset, uint32_t *data, size_t size);
+        bool flashRead(uint32_t offset, uint8_t *data, size_t size);
```

Comment:
These should have comments explaining the differences, e.g.: 
```cpp
//This call requires that offset be aligned to 4-byte boundary and size be multiple-of-4, or the call fails.
bool flashRead(uint32_t offset, uint32_t *data, size_t size);
//This allows misaligned offset and size that is not multiple-of-4, but performance slows down to handle all the corner cases.
bool flashRead(uint32_t offset, uint8_t *data, size_t size);
```
Same for Writes.